src/chat/components/ChatMessagesList.tsx

```diff
@@ -10,16 +11,30 @@ function getMessagePosition(username: string) {
 
 export default function ChatMessagesList() {
   const data = useSelector(chatMessagesSelector);
+  let timeToCheck = new Date().toJSON().slice(0, 10);
+
   return (
     <Fragment>
-      {data.map((message: ChatMessage) => {
+      {data.map((message: ChatMessage, index) => {
+        const { id, created, text, author } = message;
+        const dataToDisplay = timeToCheck;
+        const messageDate = created.slice(0, 10);
+
+        const isLastMessage = index + 1 === data.length;
+        const isDifferentDate = messageDate !== timeToCheck;
+
         return (
-          <Message
-            key={message.id}
-            position={getMessagePosition(message.author)}
-            message={message.text}
-            created={message.created}
-          />
+          <Fragment key={id}>
+            {isDifferentDate && (timeToCheck = messageDate) && (
+              <MessageDivider children={dataToDisplay} />
+            )}
+            <Message
+              position={getMessagePosition(author)}
+              message={text}
+              created={created}
+            />
+            {isLastMessage && <MessageDivider children={dataToDisplay} />}
```

Comment:
The `children ` props here is a bit confusing . If `MessageDivider` has a children prop, I would expect it to be used like so: `<MessageDivider><Child /></MessageDivider>`. Maybe we could rename children to make it more clearer?